pkg/internal/cryptoutil/utils.go

```diff
@@ -66,79 +74,66 @@ func IsChachaKeyValid(key []byte) bool {
 	return len(key) == chacha.KeySize
 }
 
-// KeyPair represents a private/public key pair
-type KeyPair struct {
-	// Priv is a private key
-	Priv []byte `json:"priv,omitempty"`
-	// Pub is a public key
-	Pub []byte `json:"pub,omitempty"`
-}
-
-// EncKeyPair represents a private/public encryption key pair
-type EncKeyPair struct {
-	KeyPair `json:"keypair,omitempty"`
-	// Alg is the encryption algorithm of keys enclosed in this key pair
-	Alg EncryptionAlgorithm `json:"alg,omitempty"`
+// KeySet contains a list of SimpleKey
+type KeySet struct {
+	ID         string      `json:"id"`
+	Keys       []SimpleKey `json:"keys"`
+	PrimaryKey SimpleKey   `json:"primarykey"`
 }
 
-// SigKeyPair represents a private/public signature (verification) key pair
-type SigKeyPair struct {
-	KeyPair `json:"keypair,omitempty"`
-	// Alg is the signature algorithm of keys enclosed in this key pair
-	Alg SignatureAlgorithm `json:"alg,omitempty"`
-}
-
-// MessagingKeys represents a pair of key pairs, one for encryption and one for signature
-// usually stored in a KMS, it helps prevent converting signing keys into encryption ones
-// TODO refactor this structure and all KeyPair handling as per issue #596
-type MessagingKeys struct {
-	*EncKeyPair `json:"enckeypair,omitempty"`
-	*SigKeyPair `json:"sigkeypair,omitempty"`
-}
-
-// isKeyPairValid is a utility function that validates a KeyPair
-func isKeyPairValid(kp KeyPair) bool {
-	if kp.Priv == nil || kp.Pub == nil {
-		return false
-	}
-
-	return true
+// SimpleKey represents a key with an ID and a value
+type SimpleKey struct {
```

Comment:
what does the prefix Simple imply?

Reply:
I will rename to Key... for some tests.. initially I didn't want to confuse it with key variable..I'll push another commit soon..

Reply:
done